synapse/tools/infra/gendc.py

```diff
@@ -0,0 +1,392 @@
+import os
+import sys
+import copy
+import stat
+import shutil
+import asyncio
+import logging
+import argparse
+
+import synapse.common as s_common
+
+import synapse.lib.config as s_config
+import synapse.lib.output as s_output
+import synapse.lib.certdir as s_certdir
+import synapse.lib.version as s_version
+
+logger = logging.getLogger(__name__)
+
+_schema_versions = '>=0.1.0,<=0.2.0'
+
+DEFAULT_DOCKER_LOGGING = {'driver': 'json-file',
+                          'options': {'max-file': '1',
+                                      'max-size': '100m'}}
+
+DEFAULT_DOCKER_NETWORK_MODE = 'host'
+DEFAULT_DOCKER_RESTART = 'unless-stopped'
+
+DOCKER_COMPOSE_DEFAULTS = {
+    'logging': DEFAULT_DOCKER_LOGGING,
+    'network_mode': DEFAULT_DOCKER_NETWORK_MODE,
+    'restart': DEFAULT_DOCKER_RESTART,
+    'labels': {},
+}
+
+infraschema = {
+    'definitions': {
+        'ahaSvc': {
+            'type': 'object',
+            'properties': {
+                'aha:network': {'type': 'string'},
+                'aha:name': {'type': 'string', 'default': 'aha'},
+                'aha:admin': {'type': 'string', 'default': 'root'},
+                'aha:listen': {'type': 'string', 'default': '0.0.0.0'},
+                'aha:port': {'type': 'integer', 'default': 27492},
+                'docker': {
+                    'type': 'object',
+                    'properties': {
+                        'image': {'type': 'string',
+                                  'default': 'vertexproject/synapse-aha:v2.x.x'},
+                        'environment': {'type': 'object', 'default': {}},
+                        'labels': {'type': 'object', 'default': {}},
+                    },
+                    'default': {
+                        'image': 'vertexproject/synapse-aha:v2.x.x',
+                        'environment': {},
+                        'labels': {},
+                    }
+                }
+            },
+            'required': [
+                'aha:network',
+            ]
+        },
+        'genericSvc': {
```

Comment:
maybe its in here and I missed it, but could there be a friendly helper to add axon (and possibly cortex) config?

Reply:
Looks like I missed GENAHAURL_* and friends